tests/by-util/test_realpath.rs

```diff
@@ -139,3 +143,50 @@ fn test_realpath_logical_mode() {
         .succeeds()
         .stdout_contains("dir1\n");
 }
+
+#[test]
+fn test_realpath_default_allows_final_non_existent() {
+    let p = Path::new("").join(GIBBERISH);
+    let (at, mut ucmd) = at_and_ucmd!();
+    let actual = ucmd.arg(p.to_str().unwrap()).run().stdout_move_str();
+    let expect = path_concat!(at.root_dir_resolved(), p.to_str().unwrap()) + "\n";
+    println!("actual: {:?}", actual);
+    println!("expect: {:?}", expect);
+    assert_eq!(actual, expect);
```

Comment:
You can use `stdout_only()` to make this test a bit simpler:
```suggestion
    ucmd.arg(p.as_os_str())
        .succeeds()
        .stdout_only(at.plus_as_string(&format!("{}\n", p.to_str().unwrap())));
```

You should be able to apply this pattern to the other tests you added as well.